src/cmd-buildextend-live

```diff
@@ -332,12 +338,33 @@ def generate_iso():
             # Assumes all files are text
             with open(srcfile) as fh:
                 buf = fh.read()
-            buf = buf.replace('@@KERNEL-ARGS@@', kargs)
+            newbuf = buf.replace('@@KERNEL-ARGS@@', kargs)
+            # if we injected kargs, also check for an embed area
+            if buf != newbuf:
+                karg_area_start = re.search(r'@@KERNEL-ARGS@@', buf)
+                buf = newbuf
+                karg_area_end = re.search(r'(#+) COREOS_KARG_EMBED_AREA\n', buf)
+                if karg_area_end is not None:
+                    kargs = buf[karg_area_start.start():karg_area_end.end()]
```

Comment:
This is modifying the first `kargs` variable so the next iteration will be replacing the wrong string into the file. Let's rename this to e.g. `file_kargs`?

Re. your comment here:

> In this case length will be less than for other karg files. I though, that it's better to have similar files (ending with COREOS_KARG_EMBED_AREA), than handing different lengths in coreos-installer.

I'm not sure if I follow. Can you expand? For all files with embed areas, we enforce a check to make sure that the cmdline is the same across all of them. So we can make that assumption here and only save the line we're interested in without the embed area in the `.cmdline` file. On the `coreos-installer` side, we know the embed area length, so we can trivially reconstruct the default layout.

Reply:
done. now it doesn't contain embed area